src/main/java/io/split/android/client/service/synchronizer/SynchronizerImpl.java

```diff
@@ -90,29 +103,31 @@ public void loadAndSynchronizeSplits() {
         List<SplitTaskBatchItem> enqueued = new ArrayList<>();
         enqueued.add(new SplitTaskBatchItem(mSplitTaskFactory.createFilterSplitsInCacheTask(), null));
         enqueued.add(new SplitTaskBatchItem(mSplitTaskFactory.createLoadSplitsTask(), mLoadLocalSplitsListener));
-        enqueued.add(new SplitTaskBatchItem(mSplitTaskFactory.createSplitsSyncTask(true, true), mSplitsSyncTaskListener));
+        enqueued.add(new SplitTaskBatchItem(new SplitTask() {
+            @NonNull
+            @Override
+            public SplitTaskExecutionInfo execute() {
+                synchronizeSplits();
+                return SplitTaskExecutionInfo.success(SplitTaskType.GENERIC_TASK);
+            }
+        }, null));
         mTaskExecutor.executeSerially(enqueued);
     }
 
     @Override
     public void synchronizeSplits(long since) {
-        SplitTask splitsUpdateTask
-                = mSplitTaskFactory.createSplitsUpdateTask(since);
-        mTaskExecutor.submit(splitsUpdateTask, null);
+        mSplitsUpdateRetryTimer.setTask(mSplitTaskFactory.createSplitsUpdateTask(since), null);
```

Comment:
recheck if it is fine to set the task here but not in the class constructor.
what if `synchronizeSplits()` is called before `synchronizeSplits(long since)`? in that case, will `mSplitsUpdateRetryTimer` have a task to run anyway?

Reply:
Thanks for the suggestion doc. Let´s discuss this in person.

Reply:
I got it javi. Thanks. They are different tasks (`mSplitsUpdateRetryTimer` and `mSplitsSyncRetryTimer`)